File: lib/to_mdx/parsers/open_document/office/document_content.rb
# frozen_string_literal: true

require_relative "../base"

module ToMdx
  module Parsers
    module Office
      class DocumentContent < Base
        attr_reader :styles, :body

        def initialize(element = nil, context = {})
          super(element, context)
          @styles = children.find { |child| child.is_a?(Office::AutomaticStyles) }
          @body = children.find { |child| child.is_a?(Office::Body) }
        end

        def to_mdx
          <<~MDX
            #{styles.to_mdx}
            #{body.to_mdx}
          MDX
        end

        def to_s
          body.to_s
        end

        register_for "office:document-content"
      end
    end
  end
end

File: src/parsers/open-document/office/document_content.ts
import dedent from "dedent-js";
import type { ParseContext } from "../../base_element.ts";
import { BaseElement } from "../../base_element.ts";
import { AutomaticStyles } from "./automatic-styles.ts";
import { Body } from "./body.ts";

export class DocumentContent extends BaseElement {
  private styles?: AutomaticStyles;
  private body?: Body;

  constructor(element: Element, context: ParseContext, parent: BaseElement | null) {
    super(element, context, parent);
    this.styles = this.children.find((child) => child instanceof AutomaticStyles);
    this.body = this.children.find((child) => child instanceof Body);
  }

  toMdx(): string {
    const stylesContent = this.styles ? this.styles.toMdx() : "";
    const bodyContent = this.body ? this.body.toMdx() : "";

    return dedent`
      ${stylesContent}
      ${bodyContent}
    `;
  }

  toString(): string {
    return this.body ? this.body.toString() : "";
  }

  static {
    BaseElement.registerFor.call(this, "office:document-content");
  }
}
